django_cloud_deploy/crash_handling/__init__.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A module handles crashes of the tool."""
+
+import os
+import platform
+import subprocess
+import sys
+import traceback
+import urllib.parse
+import webbrowser
+
+import jinja2
+
+from django_cloud_deploy.cli import io
+from django_cloud_deploy import __version__
+
+# TODO: Add an issue label for issues reported by users
+_REQUEST_URL_TEMPLATE = (
+    ('https://github.com/GoogleCloudPlatform/django-cloud-deploy/issues/new?'
+     'body={}'))
+
+
+with open(os.path.join(os.path.dirname(__file__), 'template',
+                       'issue_template.txt')) as f:
+    _ISSUE_TEMPLATE = f.read()
+
+
+def handle_crash(err: Exception, command: str,
+                 console: io.IO = io.ConsoleIO()):
+    """The tool's crashing handler.
+
+    Args:
+        err: The exception get thrown.
+        command: The command causing the exception to get thrown.
+        console: Object to use for user I/O.
+    """
+
+    # TODO: Only handle crashes caused by our code, not user's code.
+    # When deploying, our tool will run the code of user's Django project.
+    # If user's code has a bug, then the traceback will contain the directory
+    # path of their Django project. We will put traceback in a public issue,
+    # but user's might not want to put these kind of information to public,
+    # so we plan to only handle crashes caused by our code.
+    console.tell(
+        'django-cloud-deploy crashed with ({}): {}'.format(
+            type(err).__name__, err))
+
+    ans = console.ask(
+        'Would you like to file a bug against Django Cloud Deploy?\n'
+        'If you want, a browser will direct you to the tool\'s \n'
+        'Github repo. The bug will include information regarding\n'
+        'the crash and information that will help us prevent this\n'
+        'in the future.[y/N]: ')
+
+    while ans.lower() not in ['y', 'n']:
```

Comment:
I'd try to make the format more like https://github.com/GoogleCloudPlatform/django-cloud-deploy/blob/master/django_cloud_deploy/cli/prompt.py#L713

So maybe

console.tell('Your "{command}" failed due to an internal error - please try again.\n\n
                   'You can report this error by filing a bug on Github. If you agree, a '
                   'browser window will open and an Github issue will be pre-populated '
                  'with the details of this crash.')

while True:
  ans = console.ask('Would you like to file a bug return? [y/N]')
   if ans not in ['y', 'n']:
     continue
  ...

Reply:
Done.